features/step_definitions/membership_application_steps.rb

```diff
@@ -25,18 +27,28 @@
     contact_email = hash['contact_email'] && ! hash['contact_email'].empty? ?
                     hash['contact_email'] : hash[:user_email]
 
+    legacy_app = hash['legacy'] == 'true' ? true : false
+
     if (ma = user.shf_application)
 
       user.shf_application.companies << companies
 
     else
       num_categories = hash[:categories] ? 0 : 1
 
-      ma = FactoryBot.build(:shf_application,
-                            attributes.merge(user: user,
-                            contact_email: contact_email,
-                            create_company: false,
-                            num_categories: num_categories))
+      if legacy_app
+        ma = FactoryBot.build(:shf_application, :legacy_application,
+                              attributes.merge(user: user,
+                              contact_email: contact_email,
+                              create_company: false,
+                              num_categories: num_categories))
+      else
+        ma = FactoryBot.build(:shf_application,
+                              attributes.merge(user: user,
+                              contact_email: contact_email,
+                              create_company: false,
+                              num_categories: num_categories))
```

Comment:
DRY: duplicate code:
```
 attributes.merge(user: user,
                              contact_email: contact_email,
                              create_company: false,
                              num_categories: num_categories)
```
Refactor into a variable and then just use that variable if you're going to have those 2 conditional statements
ex:
```
app_attributes = attributes.merge(user: user,
                                        contact_email: contact_email,
                                        create_company: false,
                                        num_categories: num_categories)

...

ma = FactoryBot.build(:shf_application, :legacy_application, app_attributes)
...
```

Reply:
Done.